packages/scandipwa/src/util/Compare/Compare.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * ScandiPWA - Progressive Web App for Magento
+ *
+ * Copyright © Scandiweb, Inc. All rights reserved.
+ * See LICENSE for license details.
+ *
+ * @license OSL-3.0 (Open Software License ("OSL") v. 3.0)
+ * @package scandipwa/base-theme
+ * @link https://github.com/scandipwa/base-theme
+ */
+
+/**
+ *
+ * @type {string}
+ */
+export const COMPARE_UID = 'compare_uid';
+
+/**
+ *
+ * @param {String} uid
+ * @returns {void}
+ * @namespace Util/Compare/setUid
+ */
+export const setUid = (uid) => {
+    window.sessionStorage.setItem(COMPARE_UID, uid);
```

Comment:
Just wondering. Why session storage?
Data in the session storage is binded to the tab only.
If you open another tab, even with the same url (but not duplicate tab), then it will have a different session storage.
Thus you will have a different uid on all tabs.
Maybe there should be used BrowserDatabase (improved localStorage)?